pkg/clustermanager/discovery.go

```diff
@@ -139,6 +140,8 @@ func processProviderWatchEvents(ctx context.Context, w clusterprovider.Watcher,
 				// TODO: when finalizers added - cheeck the logicalcluster delete timestamp
 				// Logical cluster exists , just update its status
 				reflcluster.Status.Phase = lcv1alpha1apis.LogicalClusterPhaseReady
+				// TODO: Should we really update the config ?
```

Comment:
That's a good question. For managed clusters would we always have all the config info at the time of creation of the LogicalCluster object?  I suspect the answer is negative, but I might be wrong?

Reply:
For now let's update it. We should have a logic for "is it the same cluster", and "what to do if not"